### ISSUE TITLE
possible division by 0 in UniswapV2Library.sol

### ISSUE THREAD

**Issue by ericDeCourcy** [Uniswap/uniswap-v2-periphery#35]
It looks like [here](https://github.com/Uniswap/uniswap-v2-periphery/blob/460ffcda51492bef3b191fdaf50092e0d35b66c2/contracts/libraries/UniswapV2Library.sol#L57) there's a possibility for `denominator` to be `0`, causing a divide by `0` on the next line. Why not use the SafeMath `div` function, which provides checks and an error message for divs by `0`, instead?

**Comment by haydenadams** [comment]
I believe this is because `denominator` can never be `0` to get this point

**Comment by coffee-converter** [comment]
Or even just a adding something like:

    require(amountOut < reserveOut, 'UniswapV2Library: ...');

**Comment by ericDeCourcy** [comment]
> Or even just adding something like:
> 
> ```
> require(amountOut < reserveOut, 'UniswapV2Library: ...');
> ```

Yep! Makes logical sense too, because you should never be able to pull the entire amount of one reserve out of a uniswap market. So the `amountOut` should never be equal to or greater than `reserveOut`
